akka-discovery/src/main/scala/akka/discovery/ServiceDiscovery.scala

```diff
@@ -88,6 +89,9 @@ object ServiceDiscovery {
 @ApiMayChange
 final case class Lookup(serviceName: String, portName: Option[String], protocol: Option[String]) {
 
+  require(serviceName != null, "serviceName cannot be null")
+  require(serviceName.trim.nonEmpty, "serviceName cannot be empty")
```

Comment:
We really need a util like cats has to one line these null/empty checks, cc @patriknw unless there is one? If not I am happy to add sometime.

Reply:
What I mean is, they are important validity checks and boiler plating them is not IMO a long term solution. Thanks @renatocaval for adding them though!

Reply:
These two could be a utility function in akka.util somewhere, like `requireNonBlank(serviceName, "serviceName")`.  You still need the "serviceName" to have a nice exception message.